sensor/common/scannerclient/grpc_client.go

```diff
@@ -83,10 +176,68 @@ func (c *Client) GetImageAnalysis(ctx context.Context, image *storage.Image, cfg
 
 	log.Debugf("Received image components from local Scanner for image %s", name)
 
-	return resp, nil
+	return &ImageAnalysis{
+		ScanStatus:   resp.GetStatus(),
+		ScanNotes:    resp.GetNotes(),
+		V1Components: resp.GetComponents(),
+	}, nil
+}
+
+// Close closes and cleanup the client connection.
+func (c *v2Client) Close() error {
+	return c.conn.Close()
+}
+
+func convertIndexReportToAnalysis(ir *v4.IndexReport) *ImageAnalysis {
+	var st scannerV1.ScanStatus
+	switch ir.GetState() {
+	case "Terminal", "IndexError":
+		st = scannerV1.ScanStatus_FAILED
+	case "IndexFinished":
+		st = scannerV1.ScanStatus_SUCCEEDED
+	default:
+		st = scannerV1.ScanStatus_ANALYZING
+	}
+	return &ImageAnalysis{
+		ScanStatus: st,
+		V4Contents: ir.GetContents(),
+	}
+}
+
+func (c *v4Client) GetImageAnalysis(ctx context.Context, image *storage.Image, cfg *types.Config) (*ImageAnalysis, error) {
+	hid := fmt.Sprintf("/v4/containerimage/%s", image.GetId())
+	opts := []grpc.CallOption{grpc.WaitForReady(true)}
+	ir, err := c.GetIndexReport(ctx, &v4.GetIndexReportRequest{HashId: hid}, opts...)
+	if err == nil {
+		return convertIndexReportToAnalysis(ir), nil
+	}
+	// Check if index was "not found", so we create.
+	if s, _ := status.FromError(err); s.Code() != codes.NotFound {
+		return nil, fmt.Errorf("get index report (hashid: %s): %w", hid, err)
+	}
+	log.Infof("creating index report: %s", hid)
+	scheme := "https"
+	if cfg.Insecure {
+		scheme = "http"
+	}
+	req := &v4.CreateIndexReportRequest{
+		HashId: hid,
+		ResourceLocator: &v4.CreateIndexReportRequest_ContainerImage{
+			ContainerImage: &v4.ContainerImageLocator{
+				Url:      fmt.Sprintf("%s://%s", scheme, image.GetName().GetRemote()),
```

Comment:
Should this URL contain the registry, tag, or digest? Believe as is, the image:

`quay.io/repo/path:tag@sha256:abc`

Would yield this url:

`https://repo/path`

Reply:
Thanks for capturing this. This is a typo, and it should be `image.GetName().GetFullname()`.

Reply:
6446f056